lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -591,6 +592,74 @@ static bool parameterTypesMatch(const ValueDecl *derivedDecl,
   return true;
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+static bool overridesDifferentiableAttribute(const ValueDecl *derivedDecl,
+                                             const ValueDecl *baseDecl) {
+  ASTContext &ctx = baseDecl->getASTContext();
+  auto &diags = ctx.Diags;
+
+  auto *derivedAFD = dyn_cast<AbstractFunctionDecl>(derivedDecl);
+  auto *baseAFD = dyn_cast<AbstractFunctionDecl>(baseDecl);
+
+  if (!derivedAFD || !baseAFD)
+    return false;
+
+  auto derivedDAs = derivedAFD->getAttrs().getAttributes<DifferentiableAttr>();
+  auto baseDAs = baseAFD->getAttrs().getAttributes<DifferentiableAttr>();
+
+  // Make sure all the differentiable attributes in `baseDecl` are
+  // also declared in `derivedDecl`.
+  for (auto baseDA : baseDAs) {
+    auto baseParameters = AutoDiffIndexSubset::get(
+        ctx, baseDA->getParameterIndices()->parameters);
+    auto defined = false;
+    for (auto derivedDA : derivedDAs) {
+      auto derivedParameters = AutoDiffIndexSubset::get(
+          ctx, derivedDA->getParameterIndices()->parameters);
+      if (baseParameters->isSubsetOf(derivedParameters)) {
+        defined = true;
+      }
+    }
+    if (!defined) {
+      diags.diagnose(
+          derivedDecl,
+          diag::override_missing_differentiable,
+          derivedDecl->getFullName());
+    }
+  }
+
+  // If there is no differentiable attribute in `derivedDecl`, then
+  // overriding is not allowed.
+  if (derivedDAs.empty())
+    return false;
+
+  // Finally, go through all differentiable attributes in
+  // `derivedDecl` and check if they subsume any of the
+  // differentiable attributes in `baseDecl`.
+  auto overrides = true;
```

Comment:
Similar to the double for-loop above where you have `auto defined = false;`, you could delete this line and then on line 643 put `auto overrides = true;`